### PR TITLE
Fixes the need for a generic spider

### LINKED ISSUE #18: the gather links if statement issue
when the function tests if the file is a html file  the `response.getheader('Content-Type'):` often returns `text/html; UTF-8` so i changed the if statement to this `if 'text/html' in response.getheader('Content-Type'):`

### PR DESCRIPTION
**Pull request by BraxtonFair**
Fixes #18 and should remove the need for a generic spider and specific spiders(such as TwitterSpider, FacebookSpider, ect...)

### CODE REVIEW

File: spider.py [hunk 1]
```diff
@@ -44,8 +44,7 @@ def gather_links(page_url):
         html_string = ''
         try:
             response = urlopen(page_url)
-            if response.getheader('Content-Type') == 'text/html' or response.getheader(
-                    'content-type') == 'text/html;charset=utf-8':
+            if 'text/html' in respinse.getheader():
```
Comment (nicowaltz) [inline spider.py]: It should be response.getheader()

**Comment by buckyroberts** [comment]
It is throwing an exception when I run this code:

`getheader() missing 1 required positional argument: 'name'`

**Comment by astritzeqiri** [comment]
if 'text/html' in response.getheader('Content-Type')

**Comment by buckyroberts** [comment]
Just pushed the changes :-)

Also @BraxtonFair the getheader() function returns the value of a header, but you need to tell it which header you want the value of (which was 'Content-Type' for us).

**Pull request closed without merging**